tls_codec/src/quic_vec.rs

```diff
@@ -396,6 +397,94 @@ impl Size for &VLBytes {
     }
 }
 
+/// A wrapper struct around [`VLBytes`] that implements [`ZeroizeOnDrop`]. It
+/// behaves just like [`VLBytes`], except that it doesn't allow conversion into
+/// a [`Vec<u8>`].
+#[cfg_attr(feature = "serde", derive(SerdeSerialize, SerdeDeserialize))]
+#[derive(Clone, PartialEq, Eq, Hash, Ord, PartialOrd, Zeroize, ZeroizeOnDrop)]
+pub struct SecretVLBytes(VLBytes);
+
+impl fmt::Debug for SecretVLBytes {
```

Comment:
Can you throw all of the implementation (that makes sense) into a macro so we don't need to duplicate the code?